pkg/cloudprovider/provider/azure/get_client.go

```diff
@@ -0,0 +1,64 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-04-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-04-01/network"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+)
+
+func getIPClient(c *config) (network.PublicIPAddressesClient, error) {
+	var err error
+	ipClient := network.NewPublicIPAddressesClient(c.SubscriptionID)
+	ipClient.Authorizer, err = auth.NewClientCredentialsConfig(c.ClientID, c.ClientSecret, c.TenantID).Authorizer()
+	if err != nil {
+		return network.PublicIPAddressesClient{}, fmt.Errorf("failed to create authorizer: %s", err.Error())
+	}
+
+	return ipClient, nil
+}
+
+func getSubnetsClient(c *config) (network.SubnetsClient, error) {
+	var err error
+	ipClient := network.NewSubnetsClient(c.SubscriptionID)
+	ipClient.Authorizer, err = auth.NewClientCredentialsConfig(c.ClientID, c.ClientSecret, c.TenantID).Authorizer()
+	if err != nil {
+		return network.SubnetsClient{}, fmt.Errorf("failed to create authorizer: %s", err.Error())
```

Comment:
I think its easier if you let the func return a pointer, because then you can do `return nil, fmt.Errorf("my error")`

Reply:
fixed